### PR TITLE
chore: backport Tendermint 0.35.1 to Tenderdash 0.8

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

Backport of Tendermint 0.35.1

## What was done?

* backport of tendermint 0.35.1
* [updated mergify config](https://github.com/dashevo/tenderdash/pull/309/commits/0b0f9de795bb4f08ccdf5e62fe249bf1cea4688f)
* [updated mockery mocks](https://github.com/dashevo/tenderdash/pull/309/commits/6705471b13a06b2cba66d40558e253be8c6c06f7)
*  [chore: backport Tendermint 0.35.1 to Tenderdash 0.8 #309](https://github.com/dashevo/tenderdash/pull/309/commits/78b81116c5845554b92bd1f2cd47bd2cb8b563e0)
* [chore(ci): temporarily disable ARM build which is broken ](https://github.com/dashevo/tenderdash/pull/309/commits/8cd93e8ed22b9cbc5c24a7f0ad26e50577275e1f)
*  [test(ci): fix test test_abci_cli](https://github.com/dashevo/tenderdash/pull/309/commits/c524ee910c15c19f8295f7c3054f7dfc21de97d6)
* [chore: backport Tendermint 0.35.1 to Tenderdash 0.8 #309](https://github.com/dashevo/tenderdash/pull/309/commits/dc1c769218eb6afbd62fc8fdfad611739bba2322)
*  [build(e2e): use Go 1.18 to fix issue building curve25519-voi](https://github.com/dashevo/tenderdash/pull/309/commits/b2cb1b4c6daca9a51c63fb4f3099873622b1fc3c)

## How Has This Been Tested?

Unit tests, e2e tests

## Breaking Changes

None

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: internal/store/store.go [hunk 1]
```diff
@@ -189,6 +189,26 @@ func (bs *BlockStore) LoadBlockByHash(hash []byte) *types.Block {
...
+	}
+	if len(bz) == 0 {
+		return nil
+	}
+
+	s := string(bz)
+	height, err := strconv.ParseInt(s, 10, 64)
+
+	if err != nil {
+		panic(fmt.Sprintf("failed to extract height from %s: %v", s, err))
```
Comment (shotonoff) [inline internal/store/store.go:207]: not happy to see `panic` again :-(

**Review by shotonoff — APPROVED** [review]
LGTM 🚀
Left several minor comments, please have a look at them

File: Makefile [hunk 2]
```diff
@@ -14,9 +13,17 @@ endif
 LD_FLAGS = -X ${REPO_NAME}/version.TMCoreSemVer=$(VERSION)
 BUILD_FLAGS = -mod=readonly -ldflags "$(LD_FLAGS)"
 HTTPS_GIT := https://${REPO_NAME}.git
-DOCKER_BUF := docker run -v $(shell pwd):/workspace --workdir /workspace bufbuild/buf
+BUILD_IMAGE := ghcr.io/tendermint/docker-build-proto
+BASE_BRANCH := v0.8-dev
```
[the remark this answers is not in the document]
Comment (lklimek) [reply Makefile:17]: Done in #321

File: README.md [hunk 3]
```diff
@@ -37,8 +37,7 @@ contact us [over email](mailto:hello@interchain.berlin) or [join the chat](https
 
 ## Security
 
-To report a security vulnerability, see our [bug bounty
-program](https://hackerone.com/tendermint).
+To report a security vulnerability, see our [bug bounty program](https://hackerone.com/cosmos).
```
[the remark this answers is not in the document]
Comment (lklimek) [reply README.md:40]: I guess we don't have any, so leaving it behind.

File: README.md [hunk 4]
```diff
@@ -140,7 +138,7 @@ Additional tooling can be found in [/docs/tools](/docs/tools).
 - [Tendermint Core Blog](https://medium.com/tendermint/tagged/tendermint-core)
 - [Cosmos Blog](https://blog.cosmos.network/tendermint/home)
 
-## Join us!
+## Join us
```
[the remark this answers is not in the document]
Comment (lklimek) [reply README.md:141]: Added in #321

**Pull request merged**